src/content/docs/cheerpj3/11-guides/Networking.md

```diff
@@ -0,0 +1,87 @@
+---
+title: Networking
+description: Networking with CheerpJ
+---
+
+CheerpJ 3.0 supports different networking scenarios that are divided into two categories:
+
+- Same-origin HTTP/HTTPS requests using fetch seamlessly.
+- Generalized networking via Tailscale.
+
+## Same origin HTTP/HTTPS requests
+
+A Java application running in the browser with CheerpJ can request resources to the local server (same origin) via fetch through the HTTP/HTTPS protocols. As expected, these requests are asynchronous. Remember that for a request to be considered same-origin, it needs to match the scheme, hostname and port between the requester and the receiver.
+
+The CheerpJ runtime API implements its own HTTP/HTTPS handler using the browser `fetch()`. To use this handler you need to pass into `cheerpjInit` with the `fetch` option as shown below:
+
+```js
+async function fetchExample(url, method, postData, headers) {
+	return new Promise(function () {
+		// body
+	});
+}
+cheerpjInit({ fetch: fetchExample });
+```
+
+## Generalized networking
+
+When it comes to generalized networking such as access to private network services, peer-to-peer connections or access to the wider internet, there are some browser-enforced limitations to enhance user's security. These limitations are bound to the browser not exposing access to lower-level protocols such as UDP and TCP. The browser only allows one to perform HTTP(S) requests ruled by CORS policies.
+
+Another important scenario is the usage of WebSockets, where there is a two-way interaction between the user and the server in an event-driven manner. A WebSocket connection starts with a request from the browser via HTTP that when accepted, this connection is upgraded and does not conform to the HTTP protocol anymore. This way the user and the server can keep a connection and send bidirectional messages until this connection is closed from one of the endpoints.
```

Comment:
These 2 paragraphs add confusion in my opinion

Reply:
You're right, I have re-structured this information.